### ISSUE TITLE
Emulator crash: There was an error reading package.json

### ISSUE THREAD

**Issue by Buckwich** [firebase/firebase-tools#4641]
### [REQUIRED] Environment info

**firebase-tools:** `10.2.0` (first occurence), `11.0.1` (error still present)

**Platform:** macOS (v12.4; M1)

**Node.JS:** v16.15.0

### [REQUIRED] Test case

Setup a firebase Project with typescript support in a monorepo and use the hello world cloudFunction. 

### [REQUIRED] Steps to reproduce

Start the functions emulator. Change some code and rebuild.

### [REQUIRED] Expected behavior

In v10.1.5 this produced a warning: 
```
Error: Cannot find module '.../src/index.js'. Please verify that the package.json has a valid "main" entry
    at tryPackage (node:internal/modules/cjs/loader:353:19)
    at Function.Module._findPath (node:internal/modules/cjs/loader:566:18)
    at Function.Module._resolveFilename (node:internal/modules/cjs/loader:919:27)
    at Function.Module._load (node:internal/modules/cjs/loader:778:27)
    at Module.require (node:internal/modules/cjs/loader:1005:19)
    at require (node:internal/modules/cjs/helpers:102:18)
    at initializeRuntime (.../node_modules/firebase-tools/lib/emulator/functionsEmulatorRuntime.js:655:29)
    at processTicksAndRejections (node:internal/process/task_queues:96:5)
    at async handleMessage (.../node_modules/firebase-tools/lib/emulator/functionsEmulatorRuntime.js:699:20)

⚠  We were unable to load your functions code. (see above)
```

After the build completed the emulator would load the changes and continue to work with the changes

### [REQUIRED] Actual behavior

Since v10.2.0 the emulator exits with the following error:
```
There was an error reading .../package.json:

 .../src/index.js does not exist, can't deploy Cloud Functions
```

**Comment by joehan** [comment]
@taeold this looks like it is probably related to when we switched the emulator to use runtime delegates - could you take a look when you have a chance?

**Comment by Buckwich** [comment]
This seems to be resolved as of v11.5.0 #4826. Thanks @taeold
